python/lsst/analysis/tools/actions/plot/diaSkyPlot.py

```diff
@@ -0,0 +1,179 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("DiaSkyPanel", "DiaSkyPlot")
+
+from typing import Mapping, cast
+
+import matplotlib.pyplot as plt
+from lsst.pex.config import ConfigDictField, Field
+from matplotlib.figure import Figure
+
+from ...interfaces import KeyedData, KeyedDataSchema, PlotAction, Vector
+from .plotUtils import PanelConfig
+
+
+class DiaSkyPanel(PanelConfig):
+    """Configuration options for DiaSkyPlot panels."""
+
+    xlabel = Field[str](
+        doc="Panel x-axis label.",
+        default="RA (deg)",
+    )
+    ylabel = Field[str](
+        doc="Panel y-axis label.",
+        default="Dec (deg)",
+    )
+    invertXAxis = Field[bool](
+        doc="Invert x-axis?",
+        default=True,
+    )
+    color = Field[str](
+        doc="Point color",
+        default="C0",
+    )
+    size = Field[float](
+        doc="Point size",
+        default=5,
+    )
+    alpha = Field[float](
+        doc="Point transparency",
+        default=0.5,
+    )
+    # Eventually we might retrieve data from more columns to make the plot
+    # prettier/more information rich
+    ra = Field[str](
+        doc="Name of RA column",
+        optional=False,
+    )
+    dec = Field[str](
+        doc="Name of Dec column",
+        optional=False,
+    )
+
+
+class DiaSkyPlot(PlotAction):
+    """Generic pseudo base class for plotting DiaSources
+    (or DiaObjects) on the sky.
```

Comment:
This doc string seems awkward for some reason, is the line break in the right place?